hack/verify-golint.sh

```diff
@@ -39,13 +39,10 @@ GOLINT="${GOLINT:-$(get_golint)}"
 # we need to do this because golint ./... matches vendor...
 # we also further filter out generated k8s api code in the config packages
 # which unfortunately fails lint due to apimachinery conventions ...
-# TODO(fabrizio pandini): makes this smarter (skip only one file)
 # TODO(bentheelder): we also have to skip hack/tools for now because this is just
 # for go modules tracking of tools and has otherwise invalid imports
 # TODO(bentheelder): find a solution that does not depend on GO111MODULE="off"
 GO111MODULE="off" go list ./... | \
-  grep -v '^hack/tools$' |\
-  grep -v '^sigs.k8s.io/kind/pkg/cluster/config/v1alpha2$' | \
-  grep -v '^sigs.k8s.io/kind/pkg/cluster/config/v1alpha3$' | \
-  grep -v '^sigs.k8s.io/kind/pkg/cluster/config$' | \
+  sort -f - ${REPO_ROOT}/hack/.golint_failures ${REPO_ROOT}/hack/.golint_failures | \
```

Comment:
Because I am going to filter the output of `go list ./...` by `sort` and `uniq`.
Without repeating `${REPO_ROOT}/hack/.golint_failures` twice, if some items appear in `.golint_failures` file but not in the output of `go list ./...`, then these items will still be checked, but they don't really exist. That is wrong!